pkg/apis/core/v1/resource_tags.go

```diff
@@ -0,0 +1,74 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+// Tag adds tags to an object resource
+func Tag(ctx context.Context, a api.API, obj types.IdentifiedObject, tags ...string) error {
+	objects, err := resourceWithTagObjects(obj, tags...)
+	if err != nil {
+		return fmt.Errorf("failed to tag resource: %w", err)
```

Comment:
"failed to tag resource" is the kind of error a caller of `Tag()` would present. All errors occurring in this function (and in `Untag`) are such errors.

Please use the error message to differentiate the errors, like this one is something like "generating ResourceWithTag objects failed" and the other in this function is "API request failed".

As, from the outside perspective, the functions whole purpose is to do that API request, that second error could be returned as-is, without an added message.